server/node.go

```diff
@@ -390,6 +402,94 @@ func (n *Node) gossipCapacities() {
 	})
 }
 
+// storesScanner will walk through all the stores in the node every
+// ctx.ScanInterval and store the status in the db.
+func (n *Node) startStoresScanner(stopper *util.Stopper) {
+	stopper.RunWorker(func() {
+		start := time.Now()
+		for {
+			elapsed := time.Now().Sub(start)
+			remainingNanos := n.ctx.ScanInterval.Nanoseconds() - elapsed.Nanoseconds()
+			if remainingNanos < 0 {
+				remainingNanos = 0
+			}
+			nextIteration := time.Duration(remainingNanos)
+			log.V(6).Infof("next store scan iteration in %s", nextIteration)
+
+			select {
+			case <-time.After(nextIteration):
+				if !stopper.StartTask() {
+					continue
+				}
+
+				// Walk through all the stores on this node.
+				rangeCount := 0
+				stats := &proto.MVCCStats{}
+				storeIDs := n.lSender.GetStoreIDs()
+				accessedStoreIDs := make([]int32, 0, len(storeIDs))
+				for _, storeID := range storeIDs {
+					store, err := n.lSender.GetStore(storeID)
+					if err != nil {
+						// The rare case were a store was removed between
+						// listing the store ids and fetching their status.
+						log.Info("non-fatal error returned when scanning stores of node %v: %v", n.Descriptor.NodeID, err)
+						continue
+					}
+					storeStatus, err := store.GetStatus()
+					if err != nil {
+						log.Error(err)
+					}
+					if storeStatus == nil {
+						// The store scanner hasn't run on this node yet.
+						continue
+					}
+					accessedStoreIDs = append(accessedStoreIDs, int32(storeID))
+					rangeCount += int(storeStatus.RangeCount)
+					engine.Accumulate(stats, storeStatus.Stats)
+				}
+
+				// Store the combined stats in the db.
+				now := n.ctx.Clock.Now().WallTime
+				status := &proto.NodeStatus{
+					NodeID:     n.Descriptor.NodeID,
+					StoreIDs:   accessedStoreIDs,
+					UpdatedAt:  now,
+					StartedAt:  n.startedAt,
+					RangeCount: int32(rangeCount),
+					Stats:      *stats,
+				}
+				key := engine.NodeStatusKey(int32(n.Descriptor.NodeID))
+				if err := n.ctx.DB.Run(client.PutProtoCall(key, status)); err != nil {
+					log.Error(err)
+				}
+				// Increment iteration count.
+				n.completedScan.L.Lock()
+				n.scanCount++
+				n.completedScan.Broadcast()
+				n.completedScan.L.Unlock()
+				start = time.Now()
+				log.V(6).Infof("reset store scan iteration")
+				stopper.FinishTask()
+			case <-stopper.ShouldStop():
+				// Exit the loop.
+				return
+			}
+		}
+	})
+}
+
+// WaitForScanCompletion waits until the end of the next store scan and returns
+// the total number of scans completed so far.
```

Comment:
I assume this is only to be used from unittests? Could you unexport it if so and add a comment?